### PR TITLE
Emoji's by themselves appear 3x as large.

### LINKED ISSUE #2786: Emoji size
Rocket.Chat version: 0.25.0

The emoji are quite tiny and hard to see. One thought was whether we could show a larger emoji if the emoji is displayed on its own? Much like Skype does. Thoughts?

### PR DESCRIPTION
**Pull request by ChristianFusco**
@RocketChat/core 

Closes #2786

Added the ability to create larger emojis by editing sprites.css.
Changed renderMessageBody.js so that if a user's message contained only an emoji, it appears 3x as large.
-edited to to add issue number, oops

### CODE REVIEW

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 1]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
 	}
 
 	msg.html = message.html.replace(/\n/gm, '<br/>');
+	
+	if(singleEmojiCheck(msg))
+		msg.html = (msg.html).replace("emojione ", "big_emojione ");
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l26)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 2]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
...
+ *@para {msg} some html message
+ *@return {false} the html message either doesn't contain an emoji span, or contains other characters
+ */
+singleEmojiCheck = function(msg){
+	var str = msg.html;
+	str = str.trim();
+	//checks to make sure it's a span
+	if(!str.includes("<") || !str.includes(">"))
+		return false;
+	if(str.charAt(0) != '<' || str.charAt(str.length - 1) != '>')
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l42)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 3]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
...
+		return false;
+	if(str.indexOf(">") == str.lastIndexOf(">"))
+		return false;
+
+	//checks to make sure it's the only span
+	str = str.substring(str.indexOf("<") + 1, str.lastIndexOf(">") - 1);
+	if(!str.includes("<") || !str.includes(">"))
+		return false;
+	if(str.indexOf("<") < str.lastIndexOf(">"))
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l51)

File: packages/rocketchat-ui-message/message/renderMessageBody.js (hunk 1 above)
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Expected '{' and instead saw 'msg'.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l26)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 4]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
...
+ */
+singleEmojiCheck = function(msg){
+	var str = msg.html;
+	str = str.trim();
+	//checks to make sure it's a span
+	if(!str.includes("<") || !str.includes(">"))
+		return false;
+	if(str.charAt(0) != '<' || str.charAt(str.length - 1) != '>')
+		return false;
+	if(str.indexOf(">") == str.lastIndexOf(">"))
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l44)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 5]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
...
+/**
+ *Checks to see if a message contains only an emoji span
+ *@para {msg} some html message
+ *@return {false} the html message either doesn't contain an emoji span, or contains other characters
+ */
+singleEmojiCheck = function(msg){
+	var str = msg.html;
+	str = str.trim();
+	//checks to make sure it's a span
+	if(!str.includes("<") || !str.includes(">"))
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l40)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 6]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
...
+		return false;
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Expected '{' and instead saw 'return'.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l41)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 7]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
...
+	if(str.charAt(0) != '<' || str.charAt(str.length - 1) != '>')
+		return false;
+	if(str.indexOf(">") == str.lastIndexOf(">"))
+		return false;
+
+	//checks to make sure it's the only span
+	str = str.substring(str.indexOf("<") + 1, str.lastIndexOf(">") - 1);
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l48)

File: packages/rocketchat-ui-message/message/renderMessageBody.js (hunk 4 above)
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l44)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 8]
```diff
@@ -21,6 +21,36 @@ renderMessageBody = function(msg) {
 	}
 
 	msg.html = message.html.replace(/\n/gm, '<br/>');
+	
+	if(singleEmojiCheck(msg))
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Expected { after 'if' condition.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2486979418/issues/source?bid=3199634&fileBranchId=3266617#l25)

**Comment by ChristianFusco** [comment]
screenshot from 2016-04-28 20-11-42

Fixed spacing issues and here's a screen shot.  Sorry about the troubles, first time :)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 9]
```diff
@@ -22,5 +22,45 @@ renderMessageBody = function(msg) {
...
+		return false;
+	}
+
+	//checks to make sure it's the only span
+	str = str.substring(str.indexOf('<') + 1, str.lastIndexOf('>') - 1);
+	if (!str.includes('<') || !str.includes('>')) {
+		return false;
+	}
+
+	if (str.indexOf('<') < str.lastIndexOf('>')){
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Missing space before opening brace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2487780053/issues/source?bid=3199634&fileBranchId=3266617#l60)

File: packages/rocketchat-ui-message/message/renderMessageBody.js [hunk 10]
```diff
@@ -22,5 +22,46 @@ renderMessageBody = function(msg) {
...
+		msg.html = (msg.html).replace('emojione ', 'big_emojione ');
+	}
+
 	return msg.html;
 };
+
+/**
+ *Checks to see if a message contains only an emoji span
+ *@para {msg} some html message
+ *@return {false} the html message either doesn't
```
Comment (rodrigok) [inline packages/rocketchat-ui-message/message/renderMessageBody.js]: Codacy Issue found: [Trailing spaces not allowed.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2488056950/issues/source?bid=3199634&fileBranchId=3266617#l35)

**Pull request merged**